.github/workflows/upgrade-downgrade-e2e.yaml

```diff
@@ -0,0 +1,214 @@
+#
+# Copyright 2023 The Dapr Authors
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+name: Upgrade/Downgrade E2E tests on KinD
+
+on:
+  # Run every 12 hours on weekdays, and every 24 hours on weekends.
+  schedule:
+    - cron: "10 */12 * * 1-5"
+    - cron: "40 0 * * 0,6"
+  # Manual trigger
+  workflow_dispatch:
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  # This workflow runs our e2e tests on KinD. Tests are performed with the
+  # control plane on master and side car on the previous release, and vice
+  # versa.
+  #
+  # The e2e tests from the previous release are run against this deployment.
+  #
+  # KinD currently does not support Windows nodes so this is not intended to be
+  # a complete replacement for our AKS-based E2E workflows.
+  e2e-upgrade-downgrade:
+    name: e2e-upgrade-downgrade
+    runs-on: ubuntu-latest
+    env:
+      REGISTRY_PORT: 5000
+      REGISTRY_NAME: kind-registry
+      DAPR_REGISTRY: localhost:5000/dapr
+      DAPR_TAG: dev
+      DAPR_NAMESPACE: dapr-tests
+      DAPR_CACHE_REGISTRY: "dapre2eacr.azurecr.io"
+      PULL_POLICY: IfNotPresent
+    strategy:
+      fail-fast: false # Keep running if one leg fails.
+      matrix:
+        mode:
+        - control-plane-master
+        - dapr-sidecar-master
+    steps:
+    - name: Check out code
+      uses: actions/checkout@v3
+    - name: Get last version of dapr
+      run: |
+        echo "DAPR_LATEST_VERSION=$(curl -s https://api.github.com/repos/dapr/dapr/releases/latest | jq -r '.tag_name' | cut -c 2-)" >> $GITHUB_ENV
```

Comment:
For `workflow_dispatch` could it be possible to make this a variable that is passed as input? So manually we can test specific upgrade paths.

Reply:
Do you mean running the job control plane master vs daprd master, or do you mean overriding `DAPR_LATEST_VERSION`? If the latter, I'm not sure it makes sense to override this, since the dapr project makes no guarantees about version skew above +1 so seems like wasted effort/distracting to allow that to be run.

Reply:
I meant being able to override DAPR_LATEST_VERSION.

Although we don't have guarantees beyond N-1, I can see us wanting to test edge cases in some situations. Or maybe test a different patch release.

Reply:
Gotcha, yeah testing a specific patch def makes sense. I'll update the workflow with this capability and add a comment about what should and shouldn't be tested against